termux-shared/src/main/java/com/termux/shared/file/filesystem/FileAttributes.java

```diff
@@ -265,25 +264,35 @@ public Set<FilePermission> permissions() {
         int bits = (st_mode & UnixConstants.S_IAMB);
         HashSet<FilePermission> perms = new HashSet<>();
 
-        if ((bits & UnixConstants.S_IRUSR) > 0)
+        boolean canOwnerRead = ((bits & UnixConstants.S_IRUSR) > 0);
```

Comment:
Don't do this. Can add comments if you must, although code pretty clear and is from AOSP.